Source/Timeline/TimelineView.swift

```diff
@@ -23,21 +27,62 @@ public class TimelineView: UIView, ReusableView {
   }
 
   var eventViews = [EventView]()
-  public var layoutAttributes = [EventLayoutAttributes]() {
-    didSet {
+  public private(set) var regularLayoutAttributes = [EventLayoutAttributes]()
+  public private(set) var allDayLayoutAttributes = [EventLayoutAttributes]()
+  
+  public var layoutAttributes: [EventLayoutAttributes] {
+    set {
+      
+      // update layout attributes by separating allday from non all day events
+      allDayLayoutAttributes.removeAll()
+      regularLayoutAttributes.removeAll()
+      for anEventLayoutAttribute in newValue {
+        let eventDescriptor = anEventLayoutAttribute.descriptor
+        if eventDescriptor.isAllDay {
+          allDayLayoutAttributes.append(anEventLayoutAttribute)
+        } else {
+          regularLayoutAttributes.append(anEventLayoutAttribute)
+        }
+      }
+      
       recalculateEventLayout()
       prepareEventViews()
+      allDayView.events = allDayLayoutAttributes.map { $0.descriptor }
+      
+      //TODO: resize to height of zero if count is zero vs using hidden
+      allDayView.isHidden = allDayLayoutAttributes.count == 0
+      allDayView.scrollToBottom()
+      
       setNeedsLayout()
     }
+    get {
+      return allDayLayoutAttributes + regularLayoutAttributes
+    }
   }
   var pool = ReusePool<EventView>()
 
   var firstEventYPosition: CGFloat? {
-    return layoutAttributes.sorted{$0.frame.origin.y < $1.frame.origin.y}
+    return regularLayoutAttributes.sorted{$0.frame.origin.y < $1.frame.origin.y}
       .first?.frame.origin.y
   }
 
   lazy var nowLine: CurrentTimeIndicator = CurrentTimeIndicator()
+  
+  private var allDayViewTopConstraint: NSLayoutConstraint?
+  lazy var allDayView: AllDayView = {
+    let allDayView = AllDayView(frame: CGRect.zero)
+    
+    allDayView.translatesAutoresizingMaskIntoConstraints = false
+    addSubview(allDayView)
+
+    self.allDayViewTopConstraint = allDayView.topAnchor.constraint(equalTo: topAnchor, constant: 0)
+    self.allDayViewTopConstraint!.isActive = true
```

Comment:
You can get away with
```Swift
allDayViewTopConstraint?.isActive = true
```
and avoid crash, if constraint is `nil`.

Reply:
Soooooo! I'm sure there are different opinions about the `!` when using optionals.

Here, I feel using a `!` is ok since the line before, `line 78`, guarantees a non-nil assignment. Thus, `line 79` will never crash.


That's my reasoning of using the `!`, but I understand the risk it _may_ cause. Thus, i'll change it if you'd like me to :)

Reply:
Agree, the point is to have a consistent code style so when the other contributors would add their code, they'll be aware of such a semantic.